multierror/multierror.go

```diff
@@ -0,0 +1,28 @@
+package multierror
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Error []error
+
+func (me Error) Error() string {
+	if me == nil {
+		return ""
+	}
+
+	strs := make([]string, len(me))
+	for i, err := range me {
+		strs[i] = fmt.Sprintf("[%d] %v", i, err)
+	}
+	return strings.Join(strs, " ")
+}
+
+func (me Error) ErrorOrNil() error {
```

Comment:
The caller doesn't actually care about it being nil, all the caller wants is to take the `Error` type and yield an `error`, which may or may not be `nil`. `Error()` is the most idiomatic thing here, but that method is obviously taken via the stdlib `error` interface.